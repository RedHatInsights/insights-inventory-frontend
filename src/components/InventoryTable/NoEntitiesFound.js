import React from 'react';
import {
    EmptyStateBody,
    EmptyState,
    EmptyStateVariant,
    Title,
    EmptyStateIcon,
    Button,
    EmptyStatePrimary
} from '@patternfly/react-core';
import PropTypes from 'prop-types';
import SearchIcon from '@patternfly/react-icons/dist/esm/icons/search-icon';

/**
 * Empty state stable when no systems (or other entities) are found.
 */
const NoEntitiesFound = ({ entities, onClearAll, showIcon }) => (
    <EmptyState
        variant={EmptyStateVariant.full}
        data-ouia-component-id="empty-state"
        data-ouia-component-type="PF4/EmptyState"
        data-ouia-safe={true}
    >
        {showIcon && <EmptyStateIcon icon={SearchIcon} />}
        <Title headingLevel="h5" size="lg">
            {`No matching ${entities} found`}
        </Title>
        <EmptyStateBody>
            To continue, edit your filter settings and try again
        </EmptyStateBody>
        {onClearAll !== undefined && (
            <EmptyStatePrimary>
                <Button variant="link" onClick={onClearAll}>
                    Clear all filters
                </Button>
            </EmptyStatePrimary>
        )}
    </EmptyState>
);

NoEntitiesFound.propTypes = {
    entities: PropTypes.string,
    onClearAll: PropTypes.func,
    showIcon: PropTypes.bool
};

NoEntitiesFound.defaultProps = {
    entities: 'systems',
    showIcon: true
};

export default NoEntitiesFound;
