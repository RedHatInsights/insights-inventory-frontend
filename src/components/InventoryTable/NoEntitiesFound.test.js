import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NoEntitiesFound from './NoEntitiesFound';

describe('NoSystemsTable', () => {
    it('should render correctly - no systems', () => {
        const wrapper = mount(<NoEntitiesFound />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('h5').text()).toBe('No matching systems found');
    });

    it('should render correctly - no groups', () => {
        const wrapper = mount(<NoEntitiesFound entities='groups' />);
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.find('h5').text()).toBe('No matching groups found');
    });

    it('renders link if callback provided', () => {
        const wrapper = mount(<NoEntitiesFound onClearAll={() => 42}/>);
        expect(wrapper.find('.pf-m-link').text()).toBe('Clear all filters');
    });
});
