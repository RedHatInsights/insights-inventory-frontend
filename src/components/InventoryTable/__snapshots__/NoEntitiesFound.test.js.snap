// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSystemsTable should render correctly - no groups 1`] = `
<NoEntitiesFound
  entities="groups"
  showIcon={true}
>
  <EmptyState
    data-ouia-component-id="empty-state"
    data-ouia-component-type="PF4/EmptyState"
    data-ouia-safe={true}
    variant="full"
  >
    <div
      className="pf-c-empty-state"
      data-ouia-component-id="empty-state"
      data-ouia-component-type="PF4/EmptyState"
      data-ouia-safe={true}
    >
      <div
        className="pf-c-empty-state__content"
      >
        <EmptyStateIcon
          icon={[Function]}
        >
          <SearchIcon
            aria-hidden="true"
            className="pf-c-empty-state__icon"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden="true"
              aria-labelledby={null}
              className="pf-c-empty-state__icon"
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </SearchIcon>
        </EmptyStateIcon>
        <Title
          headingLevel="h5"
          size="lg"
        >
          <h5
            className="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-2"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe={true}
          >
            No matching groups found
          </h5>
        </Title>
        <EmptyStateBody>
          <div
            className="pf-c-empty-state__body"
          >
            To continue, edit your filter settings and try again
          </div>
        </EmptyStateBody>
      </div>
    </div>
  </EmptyState>
</NoEntitiesFound>
`;

exports[`NoSystemsTable should render correctly - no systems 1`] = `
<NoEntitiesFound
  entities="systems"
  showIcon={true}
>
  <EmptyState
    data-ouia-component-id="empty-state"
    data-ouia-component-type="PF4/EmptyState"
    data-ouia-safe={true}
    variant="full"
  >
    <div
      className="pf-c-empty-state"
      data-ouia-component-id="empty-state"
      data-ouia-component-type="PF4/EmptyState"
      data-ouia-safe={true}
    >
      <div
        className="pf-c-empty-state__content"
      >
        <EmptyStateIcon
          icon={[Function]}
        >
          <SearchIcon
            aria-hidden="true"
            className="pf-c-empty-state__icon"
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          >
            <svg
              aria-hidden="true"
              aria-labelledby={null}
              className="pf-c-empty-state__icon"
              fill="currentColor"
              height="1em"
              role="img"
              style={
                Object {
                  "verticalAlign": "-0.125em",
                }
              }
              viewBox="0 0 512 512"
              width="1em"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              />
            </svg>
          </SearchIcon>
        </EmptyStateIcon>
        <Title
          headingLevel="h5"
          size="lg"
        >
          <h5
            className="pf-c-title pf-m-lg"
            data-ouia-component-id="OUIA-Generated-Title-1"
            data-ouia-component-type="PF4/Title"
            data-ouia-safe={true}
          >
            No matching systems found
          </h5>
        </Title>
        <EmptyStateBody>
          <div
            className="pf-c-empty-state__body"
          >
            To continue, edit your filter settings and try again
          </div>
        </EmptyStateBody>
      </div>
    </div>
  </EmptyState>
</NoEntitiesFound>
`;
